src/test/kotlin/io/inventi/eventstore/eventhandler/IdempotentPersistentSubscriptionListenerTest.kt

```diff
@@ -165,6 +167,86 @@ internal class IdempotentPersistentSubscriptionListenerTest {
         verify(exactly = 1) { handler.handle() }
     }
 
+    @Test
+    fun `calls handler method with event and metadata`() {
+        // given
+        listener = listenerBuilder(EventHandlerWithMetadataImplementation(handler))
+
+        every { handler.handle() } just runs
+        eventIsUnseen(true)
+        eventShuoldBeSkipped(false)
+
+        every { handlerExtension.handle(any(), any()) } returns {}
+
+        val eventMessage = resolvedEventAWithMetadata()
+
+        // when
+        listener.onEvent(persistentSubscription, eventMessage)
+
+        // then
+        verify { handler.handle() }
+    }
+
+    @Test
+    fun `calls handler method with event and eventIds`() {
```

Comment:
can we assert if event ids are actually correct?

Reply:
Pending